android/src/fabric/java/com/brentvatne/exoplayer/events/OnTextTracksEvent.kt

```diff
@@ -0,0 +1,47 @@
+package com.brentvatne.exoplayer.events
+
+import com.brentvatne.common.Track
+import com.facebook.react.bridge.Arguments
+import com.facebook.react.bridge.WritableArray
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.uimanager.events.Event
+import com.facebook.react.uimanager.events.RCTEventEmitter
+
+class OnTextTracksEvent(viewTag: Int, private val textTracks:  ArrayList<Track>) : Event<OnTextTracksEvent>(viewTag) {
+    private val EVENT_PROP_TEXT_TRACKS = "textTracks"
+    override fun getEventName(): String {
+        return EVENT_NAME
+    }
+
+    companion object {
+        const val EVENT_NAME = "topOnTextTracks"
+    }
+
+    override fun dispatch(rctEventEmitter: RCTEventEmitter?) {
+        rctEventEmitter?.receiveEvent(viewTag, getEventName(), arrayToObject(EVENT_PROP_TEXT_TRACKS, textTracksToArray(textTracks)))
+    }
+
+    fun arrayToObject(field: String?, array: WritableArray?): WritableMap? {
+        val event = Arguments.createMap()
+        // @todo: temporarily remove put array on event callback parameter (codegen issue)
+        // event.putArray(field!!, array)
```

Comment:
still not possible with new arch ? If it is an issue with new Arch, it would be good to disable it only for old arch